src/rust/engine/fs/src/lib.rs

```diff
@@ -610,33 +626,37 @@ impl PosixFS {
   pub fn read_file(&self, file: &File) -> impl Future<Item = FileContent, Error = io::Error> {
     let path = file.path.clone();
     let path_abs = self.root.0.join(&file.path);
-    tokio_fs::File::open(path_abs)
-      .and_then(|file| {
-        tokio_io::io::read_to_end(file, Vec::new()).map(|(_file, bytes)| Bytes::from(bytes))
-      })
+    self
+      .io_pool
+      .spawn_fn(|| std::fs::read(path_abs))
+      .map(Bytes::from)
       .map(|content| FileContent { path, content })
   }
 
   pub fn read_link(&self, link: &Link) -> impl Future<Item = PathBuf, Error = io::Error> {
     let link_parent = link.0.parent().map(Path::to_owned);
     let link_abs = self.root.0.join(link.0.as_path()).to_owned();
-    tokio_fs::read_link(link_abs.clone()).and_then(move |path_buf| {
-      if path_buf.is_absolute() {
-        Err(io::Error::new(
-          io::ErrorKind::InvalidData,
-          format!("Absolute symlink: {:?}", link_abs),
-        ))
-      } else {
-        link_parent
-          .map(|parent| parent.join(path_buf))
-          .ok_or_else(|| {
-            io::Error::new(
-              io::ErrorKind::InvalidData,
-              format!("Symlink without a parent?: {:?}", link_abs),
-            )
-          })
-      }
-    })
+    let link_abs2 = link_abs.clone();
+    self
+      .io_pool
+      .spawn_fn(move || std::fs::read_link(link_abs2))
```

Comment:
Would it be possible/useful to split the baby here and do something like:
```
self.io_pool.spawn(tokio_fs::read_link(link_abs.clone()))
```
?

That feels like it would allow for a smooth API transition to the "spawn IO"/"spawn and block on"  API introduced in #8006.

Reply:
Oh, hm. Maybe that would panic due to not being on a tokio executor. But... if we literally replaced all previous calls to `self.io_pool.spawn_fn` with an explicit `spawn`...?